pkg/agent/controller/networkpolicy/packetin.go

```diff
@@ -147,8 +157,32 @@ func (c *Controller) logPacket(pktIn *ofctrl.PacketIn) error {
 		return fmt.Errorf("received error while handling packetin for NetworkPolicy: %v", err)
 	}
 
-	// Store log file
-	AntreaPolicyLogger.Printf("%s %s %s %s SRC: %s DEST: %s %d %s", ob.tableName, ob.npRef, ob.disposition, ob.ofPriority, ob.srcIP, ob.destIP, ob.pktLength, ob.protocolStr)
+	// Create 1 sec buffers for deduplicating non-Allow packet log.
+	logMsg := fmt.Sprintf("%s %s %s %s SRC: %s DEST: %s %d %s", ob.tableName, ob.npRef, ob.disposition, ob.ofPriority, ob.srcIP, ob.destIP, ob.pktLength, ob.protocolStr)
+	if ob.disposition == openflow.DispositionToString[openflow.DispositionAllow] {
+		AntreaPolicyLogger.Printf(logMsg)
+	} else {
+		_, ok := logDeduplication[logMsg]
+		// Increase count and renew buffer if duplicated within 1 sec, create buffer otherwise.
+		if ok {
+			logDeduplication[logMsg].count++
+			logDeduplication[logMsg].bufferTimer.Reset(time.Second)
+		} else {
+			record := logDedup{1, logMsg, time.Now(), time.NewTimer(time.Second)}
+			logDeduplication[logMsg] = &record
+
+			// Go routine for logging when buffer timer stops.
+			go func(r *logDedup) {
+				<-r.bufferTimer.C
+				if r.count == 1 {
+					AntreaPolicyLogger.Printf(logMsg)
```

Comment:
I think you may want to use `r.dupLogInfo` instead of `logMsg` here?

Reply:
Good point!

Reply:
BTW, do we need `dupLogInfo` prop which is the same as the key of map `logDeduplication`? We can use the key of the map as an identifier. Also each time, there will be a new `logMsg` be declared. So the `go func` could be a closure like this:
```
go func() {
	<-logDeduplication[logMsg].bufferTimer.C
	if logDeduplication[logMsg].count == 1 {
		AntreaPolicyLogger.Printf(logMsg)
	} else {
		AntreaPolicyLogger.Printf("<-%d %s-> %s", logDeduplication[logMsg].count, time.Since(logDeduplication[logMsg].initTime), logMsg)
	}
	delete(logDeduplication, logMsg)
}()
```
Not sure if it's better, just bring it up.

Reply:
I could try that, thanks for bringing up.